yt_dlp/extractor/msn.py

```diff
@@ -1,167 +1,208 @@
     }, {
-        # NBCSports Embed
-        'url': 'https://www.msn.com/en-us/money/football_nfl/week-13-preview-redskins-vs-panthers/vi-BBXsCDb',
-        'only_matching': True,
+        # Article with social embed
+        'url': 'https://www.msn.com/en-in/news/techandscience/watch-earth-sets-and-rises-behind-moon-in-breathtaking-blue-ghost-video/ar-AA1zKoAc',
+        'info_dict': {
+            'id': 'AA1zKoAc',
+            'title': 'Watch: Earth sets and rises behind Moon in breathtaking Blue Ghost video',
+            'description': 'md5:0ad51cfa77e42e7f0c46cf98a619dbbf',
+            'uploader': 'India Today',
+            'uploader_id': 'AAyFWG',
+            'tags': 'count:11',
+            'timestamp': 1740485034,
+            'upload_date': '20250225',
+            'release_timestamp': 1740484875,
+            'release_date': '20250225',
+            'modified_timestamp': 1740488561,
+            'modified_date': '20250225',
+        },
+        'playlist_count': 1,
     }]
 
     def _real_extract(self, url):
-        display_id, page_id = self._match_valid_url(url).groups()
-
-        webpage = self._download_webpage(url, display_id)
-
-        entries = []
-        for _, metadata in re.findall(r'data-metadata\s*=\s*(["\'])(?P<data>.+?)\1', webpage):
-            video = self._parse_json(unescapeHTML(metadata), display_id)
+        locale, display_id, page_id = self._match_valid_url(url).group('locale', 'display_id', 'id')
 
-            provider_id = video.get('providerId')
-            player_name = video.get('playerName')
-            if player_name and provider_id:
-                entry = None
-                if player_name == 'AOL':
-                    if provider_id.startswith('http'):
-                        provider_id = self._search_regex(
-                            r'https?://delivery\.vidible\.tv/video/redirect/([0-9a-f]{24})',
-                            provider_id, 'vidible id')
-                    entry = self.url_result(
-                        'aol-video:' + provider_id, 'Aol', provider_id)
-                elif player_name == 'Dailymotion':
-                    entry = self.url_result(
-                        'https://www.dailymotion.com/video/' + provider_id,
-                        'Dailymotion', provider_id)
-                elif player_name == 'YouTube':
-                    entry = self.url_result(
-                        provider_id, 'Youtube', provider_id)
-                elif player_name == 'NBCSports':
-                    entry = self.url_result(
-                        'http://vplayer.nbcsports.com/p/BxmELC/nbcsports_embed/select/media/' + provider_id,
-                        'NBCSportsVPlayer', provider_id)
-                if entry:
-                    entries.append(entry)
-                    continue
+        json_data = self._download_json(
+            f'https://assets.msn.com/content/view/v2/Detail/{locale}/{page_id}', page_id,
+            note='Downloading JSON metadata')
```

Comment:
'Downloading JSON metadata' is the default note
```suggestion
            f'https://assets.msn.com/content/view/v2/Detail/{locale}/{page_id}', page_id)
```